[Dominic Lim] iP

domlimm opened the pull request:
# Week 3's Individual Project Task
> P.S. It's my first PR in CS2103. 😄
<hr>

< **_HELLO THERE_** > if you're reviewing my PR! Hope you're well & thank you for doing this! 🍻 

## Contents of this PR's description
- A numbered list i.e. Features included in PR
- Code block to show a message
- A task list, like Duke!
- Tbh, I'm just filling this list up :')

Aite moving on...

Features available after merge:
1. Greet, Echo, Exit (Bye 👋🏻)
2. Add, List
3. Mark as Done
4. Todo, Event, Deadline
5. Handle Errors
6. ~~Delete~~ 🗑️ 
7. Okay imma stop here...

Now here is a list of things I need to complete in this PR:
- [x] a heading
- [x] a bullet list
- [x] a numbered list
- [x] a fenced code block (with syntax highlighting)
- [ ] a task list
- [x] an emoji
- [x] a blockquote
- [x] a hyper link
- [x] inline code
- [x] some text formatting: bold, italic, stikethrough etc.

Also, just for an `inline code`, don't `git push origin <your_branch_name> --force` please! :')

Now, a Java code block:
```java
while (true) {
    System.out.println("I love CS2103!");
}
```

Alright! Let's wrap up! Thanks again for reviewing my PR!

Feel free to reach out!
[GitHub](https://github.com/domlimm) - [LinkedIn](https://www.linkedin.com/in/dominiclimkj/)

🥂

Commit:
Add Gradle support

domlimm committed:
Add class files to ignore when staging

domlimm committed:
- Update greeting
- Change bot logo and name

domlimm committed:
- Add Input class to containerise user input
- Add enum for states
- Add echo and exit functionalities

domlimm committed:
Add add task and list tasks functionalities

domlimm committed:
- Add mark as done functionality
- Add Task class to handle task status of completion

domlimm committed:
- Separate task type into individual class
- Update print out for list and add functions
- Fix bug with toggling of mark and unmark

domlimm committed:
- Add input data for testing
- Add expected output for testing

domlimm committed:
- Add exception handling for all possible common cases
- Add Duke's exception class

domlimm committed:
Update input and output for testing

domlimm committed:
Fix bug with adding a single space char task

domlimm committed:
- Add delete functionality with exception handling
- Add extra check for mark and delete functionalities

domlimm committed:
Update test file for delete function

domlimm committed:
Update indents of switch case to follow coding standard

domlimm committed:
Update gitignore to ignore duke task file

domlimm committed:
- Add handling for file and directory
- Add displaying of tasks in duke task file
- Enhance add task function to save to duke task file

domlimm committed:
Update toggle status to update completion status in duke text file

domlimm committed:
Update delete function to remove task in duke text file

domlimm committed:
- Add date support for tasks with valid date
- Update date data type of event and deadline to LocalDate

domlimm committed:
Add new attribute to identity the different task type

domlimm committed:
Add support for searching tasks at a specific date

domlimm committed:
Merge branch 'branch-Level-7'

domlimm committed:
Merge branch 'branch-Level-8'

domlimm committed:
Fix bug: Unnecessary newline added when searching for task/s

domlimm committed:
Perform major refactors for OOP

domlimm committed:
Add packages to files in project

This is to hop on this actual/real commit and to test writing in the body of a commit to see if everything works out fine.
This commit divide classes in Duke project into packages.
Organising the classes into suitable java packages
Ah I missed a fullstop... I can't revert ugh git CLI.
Minimal and stretch goals completed!
Alright let's see if it works on GitHub.

domlimm committed:
Refactor imports to follow java's coding standard

domlimm committed:
Add JUnit test for methods in Task file

domlimm committed:
Add JUnit tests for individual sub category tasks

domlimm committed:
Add metadata for release build JAR

domlimm committed:
Add javadocs for necessary methods in classes

domlimm committed:
Refactor codes to follow coding standard

domlimm committed:
- Add find functionality
- Update gitignore to ignore vscode settings

domlimm committed:
SearchCommand: Add comments

domlimm committed:
Merge branch 'branch-A-JavaDoc'

domlimm committed:
Merge branch 'branch-A-CodingStandard'

domlimm committed:
Merge branch 'branch-Level-9'

domlimm committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

domlimm committed:
Merge branch 'branch-A-Gradle'

domlimm committed:
Update application main class name

domlimm committed:
Add config files for checkstyle

domlimm committed:
Fix checkstyle errors with main files

domlimm committed:
Fix checkstyle errors with test files

domlimm committed:
Merge branch 'branch-A-CheckStyle'

domlimm committed:
Integrate JavaFX components into Duke

bryans17 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,59 @@
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
```
I like the use of line breaks to separate logical units. Makes the code a lot more readable! Good job!

bryans17 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,108 @@
+    /**
+     * Creates new AddCommand object that receives user input.
+     * @param fullCommand fullCommand holds contents of the task to be added.
+     */
```
Perhaps insert an empty line between the description and parameter section.

bryans17 commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,28 @@
+     * @param ui ui displays feedback to the user.
+     * @param storage storage handles storage reads and writes.
+     * @throws DukeException Throw when an error occurs during command execution.
+     */
```
Perhaps insert an empty line between the description and parameter section. I noticed the same issue in several other places too.

bryans17 reviewed (commented):
Overall, a very good job following the Java Coding Standard. I really like the way you use line breaks to separate logical units in your code. Your code is very readable and clean. Only 1 minor issue with JavaDoc comments not having an empty line between the description section and parameters section.

I couldn't find any other Java Coding Standard mistakes. And once again, good job! 👍

domlimm committed:
Fix bug with starting up

domlimm committed:
- Update dialog box to use Text instead of Label to fix wrapping text
- Update images for user and bot

domlimm committed:
- Change display text's font to Roboto
- Fix checkstyle feedback

domlimm committed:
Merge branch 'branch-Level-10'

domlimm committed:
- FindCommand: Add javadocs
- Add spacing between description and others for javadocs (Code Standard)

domlimm commented:
Hey there @bryans17 ! Thank you so much for reviewing my PR and pointing out my mistake! Overlooked on that part.

I have updated all javadocs as per your suggestion and more importantly, the code standard.

Thanks so much! Happy CNY! 🍻

tiewweijian commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,28 @@
+package duke.commands;
```
Great way of packaging your classes together. Very neat and readable.

tiewweijian commented on src/main/java/duke/commands/ToggleCommand.java:
```diff
@@ -0,0 +1,94 @@
+                    : "OK, I've marked this task as not done yet:\n";
+
+            ui.output(output + taskString);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException(Constants.INVALID_INDEX_MSG);
+        } catch (IOException e) {
+            throw new DukeException(Constants.STORAGE_UPDATE_MSG);
+        } catch (IllegalArgumentException e) {
+            throw new DukeException(Constants.INVALID_MARK_MSG);
```
Great naming conventions to your constants.

tiewweijian commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,120 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.constants.Constants;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
```
Great way to separate your imports.

tiewweijian commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    /**
+     * Returns a formatted string of an Event task.
```
You might have incorrectly formatted the javadocs for this method. 
Perhaps it should look something like this
```
/**
 * @returns a formatted string of an Event task. 
 */
```

domlimm replied:
I apologise, this is slightly misleading! I should have documented both the description and the return value. Thanks for catching it!

tiewweijian reviewed (commented):
I like how you format your codes and how you organise them into packages. Very intuitive and easy to read. 👍🏻  Keep up the good work! 🥂

domlimm commented:
Hello @tiewweijian ! I appreciate the inputs! Thank you so much for reviewing my PR!

domlimm committed:
- Add title to window
- Overwrite fxml javafx version from scene builder default

brandonrhan commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,112 @@
+                task = new Deadline(taskData[0], by);
+                textToAdd = String.format(
+                        "D | %s | %s | %s",
+                        task.getCompleted() ? "1" : "0",
```
maybe you could rename this task.getCompleted() function

brandonrhan commented on src/main/java/duke/commands/ListCommand.java:
```diff
@@ -0,0 +1,43 @@
+        int length = taskList.size();
+        StringBuilder sb = new StringBuilder();
+
+        if (length == 0) {
```
this part is repeated for all subclasses of Command, maybe you could move it to Command class?

brandonrhan commented on src/main/java/duke/commands/ToggleCommand.java:
```diff
@@ -0,0 +1,98 @@
+            int index = Integer.parseInt(this.args);
+            Task updatedTask = taskList.toggleCompleted(this.isMark, --index);
+            String taskString = updatedTask.toString();
+            String updateString = "";
```
should it be updatedString wich is similar to updatedTask

brandonrhan commented on src/main/java/duke/constants/Constants.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Holds all the immutable strings used for displaying to the user.
+ */
+public final class Constants {
```
I really like this Constants class, it is so helpful!

brandonrhan commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        }
+    }
+
+    private boolean isAdd(String command) {
```
maybe should rename this function

brandonrhan commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,124 @@
+        while (sc.hasNext()) {
+            String task = sc.nextLine();
+            String[] taskData = task.split("\\s\\|\\s");
+            Boolean completed = Integer.parseInt(taskData[1]) == 1;
```
Maybe could rename this boolean as isCompleted

brandonrhan reviewed (commented):
I really like your code! The logic is very clear and easy to read. Just a suggestion maybe could us enums for different kinds of command. I have learned a lot from this!

domlimm commented:
Hey @brandonrhan ! I appreciate all the inputs. Will make the necessary changes from your suggestions! Appreciate the time you've taken to review my PR, thanks a bunch! 😃

domlimm committed:
Add application icon

domlimm committed:
Main.java: Change addAll to add

domlimm committed:
Update mainClassName to the correct value for opening jar

domlimm committed:
Add assertions for necessary classes of the program

There are several necessary classes which require assertions.

Assertions help document the important assumptions that should hold at
various points in the codebase.

domlimm committed:
Merge pull request #1 from domlimm/branch-A-Assertions

Add assertions for necessary classes of the program

domlimm committed:
Improve code quality of codebase

Some portion of the codebase requires refactoring.

To keep up with the code quality standard, refactoring has to be done.

- Update mainly the commands classes

Refactoring of codebase to follow code quality standard makes
the codes more readable to reviewers when Pull Requests (PR)
reviews are done.

domlimm committed:
Merge pull request #2 from domlimm/branch-A-CodeQuality

Improve code quality of codebase

domlimm committed:
Add streams implicitly to previously added stream

domlimm committed:
Merge pull request #3 from domlimm/branch-A-Streams

Add streams implicitly to previously added stream

domlimm committed:
Add config file for CI

domlimm committed:
Rename boolean variable based on suggestion from PR (Code Quality)

domlimm committed:
Remove IO redirection

domlimm committed:
Fix checkstyle errors

domlimm committed:
Update to search on any keyword
Currently, able to search based on task description.

Improve flexibility in searching for task/s.

Add functionality to search based on any keyword.

domlimm committed:
Refactor to use method reference instead of lambda

domlimm committed:
Merge pull request #4 from domlimm/branch-C-BetterSearch

Branch: C-BetterSearch

domlimm committed:
Fix codebase based off code quality feedback
- Task.java: Naming of boolean variable
- Duke.java: Fix header of javadocs
- Ui.java: Fix header of javadocs

domlimm committed:
Add representative screenshot of Abby

domlimm committed:
Add user guide

domlimm committed:
Set theme jekyll-theme-architect

domlimm committed:
Remove whitespaces in Ui screenshot

domlimm committed:
Remove theme for pages

domlimm committed:
Add screenshot of Abby into pages

domlimm committed:
Fix bugs
- Markdown of table
- Add space between screenshot and description

domlimm committed:
Update main header and add CI badge

domlimm committed:
Add acknowledgement for images used

domlimm committed:
Fix naming variable based on coding standard for boolean variables
